### PR TITLE
v4.10.0 - Add additional properties to annotations

### PR DESCRIPTION
**Pull request by julie-sullivan**
There are two changes in this PR.
### ClinVar - add associated scvIds, vcvIds, rcvIds to additional properties 

```
// for VCV records
additionalProperties: [
    ...,
   {
     "name": "scvIds",
     "value": "SCV00403947,SCV00403949,SCV0040393423434"
   }
]

// for SCV records
additionalProperties: [
   ...,
   {
     "name": "vcvIds",
     "value": "39478"
   }
]
```
### MNVs - other variants considered as part of phased MNV annotation

```
// variant additionalAttributes
"additionalAttributes": {
    "phasedTranscripts": {
        "attribute": {             
            "ENST0034245234": "1:343423:C:G,1:343423:CA:G",
            "ENST0034245236": "1:343423:C:G,1:343423:CA:G",   
            "ENST0034245237": "1:343423:C:G",
        }
    }
}
```

### CODE REVIEW

File: cellbase-app/src/main/java/org/opencb/cellbase/app/transform/clinical/variant/ClinVarIndexer.java [hunk 1]
```diff
@@ -266,6 +278,47 @@ private boolean updateRocksDB(AlleleLocationData alleleLocationData, PublicSetTy
...
+                property.setValue(property.getValue() + "," + String.join(",", scvAccessions));
+            }
+        }
+        if (!hasRCVIds) {
+            properties.add(new Property(null, RCVIDS, rcvAccession));
+        }
+        if (!hasSCVIds) {
+            properties.add(new Property(null, SCVIDS, String.join(",", scvAccessions)));
+        }
+        getTraitAssociation(variantAnnotation, vcvId).setAdditionalProperties(properties);
```
Comment (javild) [inline cellbase-app/src/main/java/org/opencb/cellbase/app/transform/clinical/variant/ClinVarIndexer.java:310]: @julie-sullivan this line should not be needed

File: cellbase-lib/src/test/java/org/opencb/cellbase/lib/impl/VariantAnnotationCalculatorTest.java [hunk 2]
```diff
@@ -2083,6 +2086,102 @@ public void testCheckAminoAcidChange() throws Exception {
         assertTrue(hasClinVar(variantTraitAssociation, "251327"));
     }
 
+    private static final String PHASE_DATA_URL_SEPARATOR = "\\+";
+    private static final String VARIANT_STRING_FORMAT = "\\+";
+
+    private Variant parseVariant(String variantString) {
```
Comment (javild) [inline cellbase-lib/src/test/java/org/opencb/cellbase/lib/impl/VariantAnnotationCalculatorTest.java:2092]: is this method being used anywhere?

File: cellbase-lib/src/test/java/org/opencb/cellbase/lib/impl/VariantAnnotationCalculatorTest.java [hunk 3]
```diff
@@ -2083,6 +2086,102 @@ public void testCheckAminoAcidChange() throws Exception {
...
+            throw new IllegalArgumentException("Malformed variant string " + variantString + ". "
+                    + "variantString+GT+PS expected, where variantString needs 3 or 4 fields separated by ':'. "
+                    + "Format: \"" + VARIANT_STRING_FORMAT + "\"");
+        }
+
+        return variantBuilder.build();
+    }
+
+    @Test
+    public void testKPVPMNVs() throws Exception {
```
Comment (javild) [inline cellbase-lib/src/test/java/org/opencb/cellbase/lib/impl/VariantAnnotationCalculatorTest.java:2129]: Avoid using "KPVP" in names etc. it's a very internal GeL acronym :)

**Comment by julie-sullivan** [comment]
merged into new v4.10.0 branch.

**Pull request closed without merging**